src/bin/nydus-image/inspect.rs

```diff
@@ -50,10 +53,34 @@ impl RafsInspector {
             rafs_meta,
             bootstrap: Arc::new(Mutex::new(f)),
             cur_dir_ino: root_ino,
-            parent_inoes: Vec::new(),
+            parent_inodes: Vec::new(),
+            files_parent: BTreeMap::new(),
         })
     }
 
+    // Generate the files parent inode BTreeMap for rafs v6
+    fn generate_files_parent(&mut self) -> anyhow::Result<()> {
+        let mut files_parent = std::mem::take(&mut self.files_parent);
```

Comment:
Try below since it is complex.
```
    fn generate_files_parent(&mut self) -> anyhow::Result<()> {
        let mut files_parents = BTreeMap::new();
        self.walk_dir(
            self.rafs_meta.superblock.root_ino(),
            None,
            None,
            &mut |parent, inode, _| {
                if !inode.is_dir() {
                    if let Some(parent) = parent {
                        files_parents
                            .entry(inode.ino())
                            .or_insert_with(Vec::new)
                            .push(parent.ino());
                    }
                }
                Ok(())
            },
        )?;
        self.files_parent = files_parents;
        Ok(())
    }
```

Reply:
Already replaced it. Thanks.